storage/innobase/row/row0ftsort.cc

```diff
@@ -267,6 +268,31 @@ row_fts_psort_info_init(
 				ret = FALSE;
 				goto func_exit;
 			}
+
+			/* If tablespace is encrypted, allocate additional
+			buffer for encryption/decryption. */
+			if (log_tmp_is_encrypted()) {
+
+				/* Need to align memory for O_DIRECT write */
+				psort_info[j].crypt_alloc[i] =
+					static_cast<row_merge_block_t*>(
+						ut_malloc_nokey(
+							block_size + 1024));
+
+				psort_info[j].crypt_block[i] =
+					static_cast<row_merge_block_t*>(
+						ut_align(
+							psort_info[j]
+							.crypt_alloc[i], 1024));
+
+				if (!psort_info[j].crypt_block[i]) {
```

Comment:
Is it OK to `ut_align()` on `NULL`s? If not, may be move this check just after `ut_malloc_nokey()`?

Reply:
NULL remains NULL after alignment because it is already aligned. I see no issues here.

Reply:
hmm.. however .. .`ut_align` implementation seems to disagree with me. It contains `ut_ad(ptr);`

Reply:
moved check after `ut_align`

Reply:
BTW, upstream code just few lines above has the same problem. I did not touch it.